idaes/generic_models/costing/SSLW.py

```diff
@@ -0,0 +1,1493 @@
+#################################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+"""
+Costing package based on methods from:
+
+    Process and Product Design Principles: Synthesis, Analysis, and
+    Evaluation
+    Seider, Seader, Lewin, Windagdo, 3rd Ed. John Wiley and Sons
+    Chapter 22. Cost Accounting and Capital Cost Estimation
+    22.2 Cost Indexes and Capital Investment
+
+Curently, this costing package only includes methods for capital costing of
+unit operations.
+"""
+import pyomo.environ as pyo
+
+# TODO: HX1D not supported - does not define area (has shell_area & tube_area)
+from idaes.generic_models.unit_models import (
+    Compressor,
+    CSTR,
+    Flash,
+    Heater,
+    HeatExchanger,
+    HeatExchangerNTU,
+    PFR,
+    PressureChanger,
+    Pump,
+    StoichiometricReactor,
+    Turbine)
+from idaes.generic_models.unit_models.pressure_changer import \
+    ThermodynamicAssumption
+from idaes.generic_models.unit_models.heat_exchanger \
+    import HeatExchangerFlowPattern
+
+from idaes.core.process_base import declare_process_block_class
+from idaes.core.util.exceptions import ConfigurationError
+from idaes.core.util.constants import Constants
+from idaes.core.util.math import smooth_max
+from idaes.core.util.misc import StrEnum
+
+from idaes.generic_models.costing.costing_base import (
+    FlowsheetCostingBlockData, register_idaes_currency_units)
+
+# Some more information about this module
+__author__ = "Miguel Zamarripa, Andrew Lee"
+
+
+class HXType(StrEnum):
```

Comment:
General comment on these Enums: I would (somewhat strongly) advocate the symbol and the string be the same.

Reply:
I was somewhat conflicted between backwards compatibility and more standard names here, and probably got stuck in the middle. We probably should just go all the way and make the symbol and string be the same.

I assume you would advocate they be the same down to case as well (i.e. `pumpType` should render as `"pumpType"` and not `"pump_type"`)?

Reply:
Yeah - I think I would make them the same (characters and case) -- unless there was a really good reason to make them different (`20ft` comes to mind).